### PR TITLE
SW-57507 (WBDNOS-1801): Have traceroute return a success or failure code/indicator

### PR DESCRIPTION
**Pull request by oangheluta**

### CODE REVIEW

File: traceroute/traceroute.c [hunk 1]
```diff
@@ -784,9 +784,14 @@ static void print_probe (probe *pb) {
 }
 
 
-static void print_end (void) {
+static void print_end (int result) {
 
-	printf ("\n");
+	if (result)
```
Comment (alexgelberger) [inline traceroute/traceroute.c:789]: please try to match the existing indentation style in the file

**Review by mradam-dn — APPROVED** [review]

File: traceroute/traceroute.c (hunk 1 above)
Comment (mradam-dn) [reply traceroute/traceroute.c:789]: the file indentation is inconsistent, let's merge it

File: traceroute/traceroute.c [hunk 2]
```diff
@@ -1039,8 +1045,11 @@ static void do_it (void) {
 			start++;
 		    }
 
-		    if (pb->final)
+		    if (pb->final) {
 			end = (n / probes_per_hop + 1) * probes_per_hop;
+            if (equal_addr(&pb->res, &dst_addr))
+                    dst_reached = 1;
+            }
```
Comment (alexgelberger) [inline traceroute/traceroute.c]: indentation here as well

File: traceroute/traceroute.c [hunk 3]
```diff
@@ -784,9 +784,14 @@ static void print_probe (probe *pb) {
...
 
 
-static void print_end (void) {
+static void print_end (int result) {
 
-	printf ("\n");
+	if (result)
+		printf("\nTrace successful\n");
+	else
+		printf("\nTrace failed\n");
```
Comment (alexgelberger) [inline traceroute/traceroute.c:792]: The Jira comment says to also: 
```
After 30 hops - Trace Failed
After reaching destination beofre 30 hops - Trace Successful
Abort with ctrl+c - Trace Aborted```
https://drivenets.atlassian.net/browse/SW-57507?focusedCommentId=290957

Comment (oangheluta) [reply traceroute/traceroute.c:792]: `Abort with ctrl+c - Trace Aborted` - this will be handled from CLI

**Pull request merged**